addons/sale/models/sale_order_line.py

```diff
@@ -654,6 +664,34 @@ def _compute_qty_delivered(self):
         for so_line in lines_by_analytic:
             so_line.qty_delivered = mapping.get(so_line.id or so_line._origin.id, 0.0)
 
+    def _get_downpayment_state(self):
+        self.ensure_one()
+
+        if self.display_type:
+            return ""
+
+        invoice_lines = self._get_invoice_lines()
+        if all(line.parent_state == "draft" for line in invoice_lines):
+            return "draft"
+        if all(line.parent_state == "cancel" for line in invoice_lines):
+            return "cancel"
+        if all(line.parent_state == "posted" for line in invoice_lines):
+            return "posted"
+
+        return ""
+
+    def _show_downpayment_section(self):
+        self.ensure_one()
+
+        if (
+            self.display_type == 'line_section' and
+            self.is_downpayment and
+            self.order_id
+        ):
+            return self.order_id._count_downpayment_lines() > 0
+
+        return True
```

Comment:
Method can be removed with my proposal above, there is no need to have a separate method for this imho.

Reply:
Is it used anywhere now?